components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -1165,16 +1165,25 @@ func (p *pg) ListRulesForProject(ctx context.Context, projectID string) ([]*v2.R
 		return nil, v2.RulesStatusError, err
 	}
 
-	tx, err := p.db.BeginTx(ctx, nil)
-	if err != nil {
-		return nil, v2.RulesStatusError, p.processError(err)
+	// in our other APIs we use a a postgres query to do filtering
+	// however in this case, we can't automatically assume NoRows means NotFound
+	// because we want to differentiate between a project that is not in the project filter
+	// and a project that has no rules
```

Comment:
A project that has no rules will still return an empty array instead of NotFound-- unless the user doesn't have permission to see that project, in which case they will see NotFound.

Reply:
we do want a project that has no rules to return an empty array.
the issue we found was that when we do the filtering check [here](https://github.com/chef/automate/blob/master/components/authz-service/storage/postgres/migration/sql/54_use_project_db_id_in_references_of_project_rules.up.sql#L72), we found we return an empty array for both cases when the project has no rules and the project is not found.

maybe you have some ideas how to keep it in the query? i agree, i don't like the special casing